libvast/src/system/partition.cpp

```diff
@@ -704,6 +753,29 @@ partition_actor::behavior_type passive_partition(
       auto eval = self->spawn(evaluator, expr, triples);
       return self->delegate(eval, client);
     },
+    [=](atom::status, status_verbosity /*v*/) -> caf::config_value::dictionary {
+      const auto& st = self->state;
+      caf::settings result;
+      caf::put(result, "size", st.partition_chunk->size());
+      size_t mem_indexers = 0;
+      for (size_t i = 0; i < st.indexers.size(); ++i) {
+        if (st.indexers[i])
+          mem_indexers
+            += sizeof(indexer_state)
+               + st.flatbuffer->indexes()->Get(i)->index()->data()->size();
+      }
+      caf::put(result, "memory-usage-indexers", mem_indexers);
+      auto x = st.partition_chunk->incore();
+      if (!x) {
+        caf::put(result, "memory-usage-incore", render(x.error()));
```

Comment:
We shouldn't render errors inline with results. Shouldn't the whole handler reply with the error instead, possibly enriched with some information about the indexer? Then we can print the error at the outermost status request.

Reply:
Then the failure of one indexer would shadow the data from the whole component. I reusing the key in a "value or error" style is best.
The alternatives are:
 - omitting the key, and inserting the error string under the `error` key. 
 - Setting the key to a neutral value and adding the error under the `error` key.
Both have 2 drawbacks: If 2 keys in the same JSON node can fail to report, we can only insert the error for 1 of them. Second, scripts for automatic evaluation might use a `select()` mechanism to retrieve the key from the object. This would silently fail and lead to wrong information in the generated report.

Reply:
Let's put this on hold until we implement `ch20023`.